pyrefly/lib/test/flow_looping.rs

```diff
@@ -685,3 +685,20 @@ for i in range(10):
     a = d.get('x', a)
     "#,
 );
+
+// Test for https://github.com/facebook/pyrefly/issues/1565
+testcase!(
+    bug = "False positive: int is not assignable to int with nested while loops",
```

Comment:
We generally only use `bug = "..."` to track bugs that aren't fixed; in this case it looks like we fixed the issue. That's just for future reference - I'll patch it up since I have to import the PR anyway in order to merge